sv.c

```diff
@@ -4547,6 +4547,16 @@ Perl_sv_setsv_flags(pTHX_ SV *dsv, SV* ssv, const I32 flags)
         }
         if (sflags & SVp_NOK) {
             SvNV_set(dsv, SvNVX(ssv));
+            if ((sflags & SVf_NOK) && !(sflags & SVf_POK)) {
```

Comment:
maybe I am being predantic but I would have written that as 
```
if ((sflags & (SVf_NOK|SVf_POK)) == SVf_NOK) { 
```

Reply:
This is copying the exact code in the block after it, just with different flags. I don't want to change the IV block when not needed, but I also want this to match in terms of style.

I think this entire code path is in need of refactoring, so I would prefer to make it obvious that this is meant to match its sibling block.

Reply:
Fair enough. Ill pretend you have promised to the refactor.